server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/HistogramAggregationBuilder.java

```diff
@@ -88,6 +89,14 @@ public static HistogramAggregationBuilder parse(String aggregationName, XContent
         return PARSER.parse(parser, new HistogramAggregationBuilder(aggregationName), null);
     }
 
+    private static boolean wasRegistered = false;
```

Comment:
Do we expect this to be temporary?  I think based on our conversations it is (until we sort out why the tests are re-registering repeatedly)?  Just asking because it'd be nice if we could keep all the aggs from having to implement this flag (easy to forget).

If we expect it to be permanent, we should probably use a SetOnce or something similar to ensure we don't change it.

Reply:
I hope this is temporary, but I'm going to go ahead and slap a SetOnce on it anyway.  Honestly, I'm shocked this worked as is without being set to volatile, so that seems like cheap insurance overall.  I'll add a line item to the meta ticket to revisit this so we don't forget (and probably a TODO in the code as well)